Unable to build after uninstalling tap and @types/tap from a project

I'm trying to transition to using the node:test internal module in my project. However, when I uninstall tap and @types/tap, the project stops building, and tsc throws an error saying it cannot find the module 'tap', which is imported in src/types.ts.

## Steps to reproduce:

1. Start from a project that builds successfully when tap and @types/tap are installed:

```shell
$~/quick-test> rm -rf tsconfig.tsbuildinfo lib && npm run build

> build
> npx tsc --build ./

$~/quick-test>
```

2. Uninstall tap and @types/tap and rebuild the project:

```shell
$~/quick-test> npm uninstall tap @types/tap

removed 310 packages, and audited 79 packages in 1s

6 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
$~/quick-test> rm -rf tsconfig.tsbuildinfo lib && npm run build

> build
> npx tsc --build ./

node_modules/@mojojs/core/lib/types.d.ts:10:27 - error TS2307: Cannot find module 'tap' or its corresponding type declarations.

10 import type { Test } from 'tap';
                             ~~~~~

node_modules/@mojojs/core/lib/user-agent/test.d.ts:6:27 - error TS2307: Cannot find module 'tap' or its corresponding type declarations.

6 import type { Test } from 'tap';
                            ~~~~~


Found 2 errors.

```

## Expected behavior:

The project should build without errors after uninstalling tap and @types/tap.

## Actual behavior:

The project fails to build, as tsc cannot find the tap module due to the Test type being explicitly imported in src/types.ts.

--- a/src/types.ts
+++ b/src/types.ts
@@ -7,8 +7,23 @@ import type {Agent} from 'node:http';
 import type {Readable, Stream} from 'node:stream';
 import type {URL} from 'node:url';
 import type {InspectOptions} from 'node:util';
-import type {Test} from 'tap';
 import type {CookieJar} from 'tough-cookie';
+
+export interface TestContext {
+  equal(actual: any, expected: any, message?: string): void;
+  not(actual: any, expected: any, message?: string): void;
+  notMatch(actual: any, pattern: RegExp, message?: string): void;
+  match(actual: any, pattern: RegExp, message?: string): void;
+  same(actual: any, expected: any, message?: string): void;
+  plan(count: number): void;
+  skip(message: string): void;
+  todo(message: string): void;
+  ok(value: any, message?: string): void;
+  end(): void;
+
+  beforeEach(callback: (t: TestContext) => Promise<void>): void;
+  afterEach(callback: () => Promise<void>): void;
+}
 
 export type {JSONValue} from '@mojojs/util';
 
@@ -221,7 +236,7 @@ export interface UserAgentWebSocketOptions extends SharedUserAgentRequestOptions
 
 export type URLTarget = string | [string, MojoURLOptions];
 
-export type TestUserAgentOptions = UserAgentOptions & {tap?: Test};
+export type TestUserAgentOptions = UserAgentOptions & {tap?: TestContext};
 
 export interface ValidationError {
   instancePath: string;

--- a/src/user-agent/test.ts
+++ b/src/user-agent/test.ts
@@ -3,13 +3,13 @@ import type {App} from '../app.js';
 import type {
   JSONValue,
   ServerOptions,
+  TestContext,
   TestUserAgentOptions,
   UserAgentRequestOptions,
   UserAgentWebSocketOptions
 } from '../types.js';
 import type {WebSocket} from '../websocket.js';
 import type {URL} from 'node:url';
-import type {Test} from 'tap';
 import assert from 'node:assert/strict';
 import {on} from 'node:events';
 import {MockUserAgent} from './mock.js';
@@ -23,7 +23,7 @@ type SkipFunction = (...args: any[]) => any;
 // Helper function to add required `TestUserAgent` assert methods
 // that are missing in a `TAP` instance. This is as an intended side effect
 // to ensure compatibility with `TestUserAgent` class below
-function addNodeAssertMethods(tapInstance: Test): void {
+function addNodeAssertMethods(tapInstance: TestContext): void {
   (tapInstance as any).strictEqual = tapInstance.equal.bind(tapInstance);
   (tapInstance as any).notStrictEqual = tapInstance.not.bind(tapInstance);
   (tapInstance as any).doesNotMatch = tapInstance.notMatch.bind(tapInstance);
@@ -39,7 +39,7 @@ export class TestUserAgent extends MockUserAgent {
    */
   body: Buffer = Buffer.from('');
 
-  _assert: typeof assert | Test | undefined = undefined;
+  _assert: typeof assert | TestContext | undefined = undefined;
   _dom: DOM | undefined = undefined;
   _finished: [number, string] | null | undefined = undefined;
   _messages: AsyncIterableIterator<JSONValue> | undefined = undefined;
@@ -372,13 +372,13 @@ export class TestUserAgent extends MockUserAgent {
     return this._dom;
   }
 
-  _prepareTap(tap: Test): void {
+  _prepareTap(tap: TestContext): void {
     addNodeAssertMethods(tap);
     this._assert = tap;
     const subtests = [tap];
     const assert = this._assert;
 
-    assert.beforeEach(async t => {
+    assert.beforeEach(async (t: TestContext) => {
       addNodeAssertMethods(t);
       subtests.push(t);
       this._assert = t;

